packages/integrations/vercel/src/analytics.ts

```diff
@@ -0,0 +1,69 @@
+import { inject } from '@vercel/analytics';
+import type { Metric } from 'web-vitals';
+import { getCLS, getFCP, getFID, getLCP, getTTFB } from 'web-vitals';
+
+const vitalsUrl = 'https://vitals.vercel-analytics.com/v1/vitals';
+
+type Options = { path: string; analyticsId: string; debug?: boolean };
+
+const getConnectionSpeed = () => {
+	return 'connection' in navigator &&
+		navigator['connection'] &&
+		'effectiveType' in (navigator['connection'] as unknown as { effectiveType: string })
+		? (navigator['connection'] as unknown as { effectiveType: string })['effectiveType']
+		: '';
+};
+
+const sendToAnalytics = (metric: Metric, options: Options) => {
+	const body = {
+		dsn: options.analyticsId,
+		id: metric.id,
+		page: options.path,
+		href: location.href,
+		event_name: metric.name,
+		value: metric.value.toString(),
+		speed: getConnectionSpeed(),
+	};
+
+	if (options.debug) {
+		// eslint-disable-next-line no-console
+		console.log('[Analytics]', metric.name, JSON.stringify(body, null, 2));
+	}
```

Comment:
We’re not really exposing a way for users to pass `debug: true` so maybe we can remove this and the `debug?` field in the `Options` type?

Reply:
This function is provided by vercel. I made no changes. I can try to simplify